server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -132,12 +132,20 @@
     public MapperService(IndexSettings indexSettings, IndexAnalyzers indexAnalyzers, NamedXContentRegistry xContentRegistry,
                          SimilarityService similarityService, MapperRegistry mapperRegistry,
                          Supplier<QueryShardContext> queryShardContextSupplier, BooleanSupplier idFieldDataEnabled) {
+        this(indexSettings, indexAnalyzers, xContentRegistry, similarityService, mapperRegistry, queryShardContextSupplier,
+            idFieldDataEnabled, null);
+    }
+
+    public MapperService(IndexSettings indexSettings, IndexAnalyzers indexAnalyzers, NamedXContentRegistry xContentRegistry,
```

Comment:
Could we delete the extra `MapperService` constructor above, to make it harder to forget to pass the timestamp field? It looks like it's only used in tests and for simulating a merge.